src/streamed/state.ts

```diff
@@ -114,6 +114,26 @@ export class AuctionState {
     private readonly options: OrderbookOptions,
   ) {}
 
+  /**
+   * Creates a copy of the auction state that can apply events independently
+   * without modifying the original state.
+   */
+  public copy(): AuctionState {
```

Comment:
Cool! Can we add a unit test that this actually creates a deep copy? It seems to work because bigint seems to be an 
 immutable type (I think with BN we would have to also copy the data of the maps).

Reply:
Let me know if you think the test is sufficient.

Reply:
Perfect, thanks!